tidb-cloud/setup-self-hosted-kafka-private-service-connect.md

```diff
@@ -0,0 +1,704 @@
+
+2. Go to the detail page of the node in kafka-proxy-ig. Click **SSH** to log in to the node. Download the binaries:
+
+    ```shell
+    # You can choose another version 
+    wget https://github.com/grepplabs/kafka-proxy/releases/download/v0.3.11/kafka-proxy-v0.3.11-linux-amd64.tar.gz
+    tar -zxf kafka-proxy-v0.3.11-linux-amd64.tar.gz
+    ```
+
+3. Run Kafka-proxy and connect to Kafka brokers.
+
+    ```shell
+    # There are three kinds of parameters that need to feed to the Kafka-proxy
+    # 1. --bootstrap-server-mapping defines the bootstrap mapping. Suggest that you configure three mappings, one for each zone for resilience.
+    #   a) Kafka broker address; 
+    #   b) Local address for the broker in Kafka-proxy; 
+    #   c) Advertised listener for the broker if Kafka clients bootstrap from Kafka-proxy
+    # 2. --dynamic-sequential-min-port defines the start port of the random mapping for others brokers
+    # 3. --dynamic-advertised-listener defines advertised listener address for others brokers based on the pattern got from "Prerequisites" section
+    #   a) The pattern: <broker_id>.abc.us-west1.gcp.3199745.tidbcloud.com:<port>
+    #   b) Replace <broker_id> to fixed lower case string, for example "brokers". You can use your own string, but it's MUST. This will help TiDB Cloud route requests properly.
+    #   c) Remove ":<port>"
+    #   d) The advertised listener address would be: brokers.abc.us-west1.gcp.3199745.tidbcloud.com
+    ./kafka-proxy server \
+            --bootstrap-server-mapping "{address_of_broker1},0.0.0.0:9092,b1.abc.us-west1.gcp.3199745.tidbcloud.com:9092" \
+            --bootstrap-server-mapping "{address_of_broker2},0.0.0.0:9093,b2.abc.us-west1.gcp.3199745.tidbcloud.com:9093" \
+            --bootstrap-server-mapping "{address_of_broker3},0.0.0.0:9094,b3.abc.us-west1.gcp.3199745.tidbcloud.com:9094" \
+            --dynamic-sequential-min-port=9095 \
+            --dynamic-advertised-listener=brokers.abc.us-west1.gcp.3199745.tidbcloud.com > ./kafka_proxy.log 2>&1 &
+    ```
+
+4. Test bootstrap in Kafka-proxy node.
+
+    ```shell
+    # Download Kafka and OpenJDK, and then extract the files. You can choose the binary version based on your preference.
+    wget https://downloads.apache.org/kafka/3.7.1/kafka_2.13-3.7.1.tgz
+    tar -zxf kafka_2.13-3.7.1.tgz
+    wget https://download.java.net/java/GA/jdk22.0.2/c9ecb94cd31b495da20a27d4581645e8/9/GPL/openjdk-22.0.2_linux-x64_bin.tar.gz
+    tar -zxf openjdk-22.0.2_linux-x64_bin.tar.gz
+
+    export JAVA_HOME=~/jdk-22.0.2
+
+    ./kafka_2.13-3.7.1/bin/kafka-broker-api-versions.sh --bootstrap-server 0.0.0.0:9092
+    # Expected output(lines in tail), order may be different.
+    # There are exceptions or errors since advertised listeners can not be resolved in your network.
```

Comment:
```suggestion
    # Expected output of the last few lines (the actual order might be different)
    # There might be exceptions or errors because advertised listeners cannot be resolved in your network.
```